File: 058_Length_of_Last_Word/058_1.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        if not s:
            return 0
        count = 0
        for i in reversed(range(len(s))):
            if s[i] == ' ':
                if count == 0:
                    continue
                break
            count += 1
        return count

File: 058_Length_of_Last_Word/058_1.c
#include <stdio.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>
#include <math.h>
#include <stdint.h>
#include <string.h>

int lengthOfLastWord(char * s){
    size_t len = strlen(s);
    if (len == 0) return 0;
    int count = 0;
    int i = len - 1;
    while (i >= 0 && s[i] == ' ') i--;
    while (i >= 0 && s[i] != ' ') {
        count++;
        i--;
    }
    return count;
}


int main(void) {
    char *s = "World World";
    printf("%d\n", lengthOfLastWord(s));
}

